### PR TITLE
Added retries for messages in the queue

### LINKED ISSUE #936: Consider informing user and/or dropping HTTP POSTs on 4xx errors
As shortly discussed and reported in https://github.com/owntracks/ios/issues/663 we might consider informing the user of HTTP `4xx` codes upon HTTP POST requests, and optionally dropping the payload completely until the client and server have solved the issue.

As seen in the case of our client with a Traccar server, not deleting a `_type: dump` request will basically disable the client forever.

Note that this affects HTTP only.

### PR DESCRIPTION
**Pull request by growse**
Fixes #936

Current implementation sets a default retry of 10, but overrides it to 10,000 for `MessageLocation` types. The idea here is that all messages should have some retries, but we probably don't want to throw away locations so easily. Devices can be out of network coverage for a long period of time, and a failed send will be retried every minute. 10,000 retries buys up to a week which *feels* like a good threshold, but we might want to do something different.

### CODE REVIEW

File: project/app/src/main/java/org/owntracks/android/model/messages/MessageLocation.kt [hunk 1]
```diff
@@ -10,6 +10,9 @@ import java.lang.ref.WeakReference
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
 open class MessageLocation(private val dep: MessageWithCreatedAt = MessageCreatedAtNow(RealClock())) : MessageBase(), MessageWithCreatedAt by dep {
+    @JsonIgnore
+    override val numberOfRetries: Int = 10000 // This should last a week at 1 attempt per minute
```
Comment (jpmens) [inline project/app/src/main/java/org/owntracks/android/model/messages/MessageLocation.kt:14]: Do you want to make this a full week? At once/minute, 10080 is a week.

**Review by jpmens — APPROVED** [review]
Looks good to me.

File: project/app/src/main/java/org/owntracks/android/model/messages/MessageLocation.kt (hunk 1 above)
Comment (growse) [reply project/app/src/main/java/org/owntracks/android/model/messages/MessageLocation.kt:14]: I rounded down :p

I guess we should be true to a human spec, not a base-10 arbitrary spec.

**Comment by growse** [comment]
I should write a test.

**Pull request merged**